src/libraries/System.Private.Uri/src/System/IriHelper.cs

```diff
@@ -47,6 +48,31 @@ internal static bool CheckIriUnicodeRange(char highSurr, char lowSurr, out bool
             return false;
         }
 
+        internal static bool CheckIriUnicodeRange(uint value, bool isQuery)
+        {
+            if (value <= 0xFFFF)
+            {
+                return IsInInclusiveRange(value, '\u00A0', '\uD7FF')
+                    || IsInInclusiveRange(value, '\uF900', '\uFDCF')
+                    || IsInInclusiveRange(value, '\uFDF0', '\uFFEF')
+                    || (isQuery && IsInInclusiveRange(value, '\uE000', '\uF8FF'));
+            }
+            else
+            {
+                return ((value & 0xFFFF) < 0xFFFE)
+                    && ((value - 0xE0000) >= (0xE1000 - 0xE0000))
```

Comment:
Nit: `!IsInInclusiveRange(value, 0xE0000, 0xE0FFF)` is more readable than `((value - 0xE0000) >= (0xE1000 - 0xE0000))`.